src/palace/manager/metadata_layer/container.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from pydantic import BaseModel, ConfigDict
+from sqlalchemy.orm import Session
+
+from palace.manager.metadata_layer.identifier import IdentifierData
+from palace.manager.sqlalchemy.model.datasource import DataSource
+from palace.manager.sqlalchemy.model.identifier import Identifier
+from palace.manager.util.log import LoggerMixin
+
+
+class BaseDataContainer(BaseModel, LoggerMixin):
+    model_config = ConfigDict(
+        frozen=False,
+        validate_assignment=True,
```

Comment:
We may eventually want to disable this, if its having negative performance impacts. Since it causes the validators to run every time the data in the container changes. 

This makes sure that some of the data manipulation that we do in the validators is always performed. So far the performance seems okay, but I wanted to note this.

Reply:
Should we put a note to that effect in the code itself?

Reply:
Done